src/lib/database/store.ts

```diff
@@ -2,45 +2,116 @@ import { immer } from "zustand/middleware/immer"
 import { createStore, type StoreApi } from "zustand/vanilla"
 import { hoist } from "zustand-hoist"
 
-import * as m from "lib/models.ts"
-
 import type { Database, ZustandDatabase } from "./schema.ts"
 
 export const createDatabase = (): ZustandDatabase => {
   return hoist<StoreApi<Database>>(createStore(initializer))
 }
 
+const getImageIdFromImageUrl = (url: string): string => {
+  const image_url = new URL(url)
+  const image_id = image_url.searchParams.get("image_id")
+  if (!image_id) {
+    throw new Error(`No image_id in ${url}`)
+  }
+
+  return image_id
+}
+
 const initializer = immer<Database>((set, get) => ({
-  _id: 0,
-  things: [],
+  manufacturers: [],
+  device_models: [],
   vercel_protection_secret: "abc123",
+  external_image_proxy_endpoint:
+    "https://connect.getseam.com/internal/devicedb_image_proxy",
+
+  addManufacturerFromLiveApi(manufacturer) {
```

Comment:
Just curious, why is there a difference between adding a "live" manufacturer etc vs a non-live one? Shouldn't it be the same at this point?

Reply:
live/publicly mapped objects have a different shape than stored objects (i.e. the image_url -> image_id mapping)